NewPass/PasswordGeneratorViewController.swift

```diff
@@ -114,12 +136,23 @@ class PasswordGeneratorViewController: UIViewController {
         // Update labels
         passwordLabel1.attributedText = passwordString
     }
-    
-    private func updateLabelsWithAnimation() {
+
+    ///
+    /// This function does a lot. It saves the new password as a variable at the top of this file, and also handles the animation of swapping out the
+    /// old password for the new one in the UI. Plus, if the user still is holding down the button, this function will generate _another_ new password
+    /// and call itself again.
+    ///
+    /// Perhaps consider renaming this function, as you have one function updateLabelsForPassword()
+    /// that sounds like it should do something similar, but this one really does more than just update the lables... Not a big deal but still. 😀
+    ///
+    /// - Parameter newPassword: the new password to update to, both on the lable and in memory
+    ///
+    private func updateLabels(with newPassword: NSAttributedString, animationDuration: Double = 0.3) { // default to 0.3 if no duration provided
```

Comment:
I messed with this exact idea in the past (setting a default value for the duration) but it still was not animating how I wanted it to. I think it was because of the sprint damping and velocity, perhaps? I'll mess around a little more and see if the animations look as intended. I agree this is a much better solution if the animation stays the same.

Reply:
Ooohhhh. I see. the animations were slightly different in more ways than just the timing. Yea my PR changes the behavior, which isn't what you want. My bad. 

You can still probably achieve this without duplicating the entire animation logic again as a separate almost identical function. You can probably just play with it to figure out what you want the `usingSpringWithDamping` and `initialSpringVelocity` to be on "round 1" and on "round 2", and then pass those values in manually at the call sites like my PR does with `duration`.